File: backend-java/src/main/java/com/aiproject/article/dto/PageResponse.java
package com.aiproject.article.dto;

import lombok.AllArgsConstructor;
import lombok.Data;

import java.util.List;

/**
 * 分页响应，对应契约 GET /api/v1/articles 的 data 结构。
 */
@Data
@AllArgsConstructor
public class PageResponse<T> {

    private List<T> items;
    private long total;
    private int page;
    private int pageSize;
}

File: backend-dotnet/AiProject.Api/DTOs/PageResponse.cs
namespace AiProject.Api.DTOs;

/// <summary>
/// 分页响应，对应契约 GET /api/v1/articles 的 data 结构。
/// </summary>
public class PageResponse<T>
{
    public List<T> Items { get; set; } = new();
    public long Total { get; set; }
    public int Page { get; set; }
    public int PageSize { get; set; }
}
